### PR TITLE
added typechecking for all ts files

### LINKED ISSUE #6436: Add all TS files to front type checker.
## Scope & Context

We often have problems with the CI for typecheck because stories and test files aren't tested by the typescript process during regular development.

This makes us lose time during code review and create work that could just be solved easily when the developer sees warnings in the console.

## Technical inputs

We want to integrate all typescript files in the typescript process that Vite uses for running the frontend, so that even test files TS errors appear during the normal development flow.

### PR DESCRIPTION
**Pull request by Mihan786Chistie**
Fixes: #6436 

Changes made: 

- Added typecheck step before twenty-ui build to check stories TS errors
- Added a tsconfig.dev.json to add stories and tests to typecheking when in dev mode
- Added tsconfig.dev.json to storybook dev command of twenty-ui to typecheck stories while developing
- Fixed twenty-ui stories that were broken

- Added a serve command to serve front build
- Fixed unit test from another PR

### CODE REVIEW

File: .github/workflows/ci-front.yaml [hunk 1]
```diff
@@ -139,4 +139,7 @@ jobs:
         uses: ./.github/workflows/actions/nx-affected
         with:
           tag: scope:frontend
-          tasks: ${{ matrix.task }}
\ No newline at end of file
+          tasks: ${{ matrix.task }}
+      - name: TypeScript Type Check
```
Comment (lucasbordeau) [inline .github/workflows/ci-front.yaml:143]: That's not necessary we already have a CI task for that

File: packages/twenty-front/tsconfig.json [hunk 2]
```diff
@@ -28,7 +28,14 @@
...
   },
   "files": [],
-  "include": [],
+  "include": [
+    "src/**/*.ts",
+    "src/**/*.tsx",
+    "src/**/*.stories.ts",
+    "src/**/*.stories.tsx",
+    "src/**/*.test.ts",
+    "src/**/*.test.tsx"
```
[the remark this answers is not in the document]
Comment (lucasbordeau) [reply packages/twenty-front/tsconfig.json:37]: I think the solution would involve `tsconfig.app.json`, maybe we should just remove the excluded array ? 

Please test by modifying a TS type that is both present in a .tsx file and a story file, you dev env in VS Code should return an error for the story, while it is not doing it right now.

File: .github/workflows/ci-front.yaml (hunk 1 above)
Comment (Mihan786Chistie) [reply .github/workflows/ci-front.yaml:143]: Ok

File: packages/twenty-front/tsconfig.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mihan786Chistie) [reply packages/twenty-front/tsconfig.json:37]: Ok I will try that

**Comment by Mihan786Chistie** [comment]
@lucasbordeau Please check the recent fixes

**Comment by Mihan786Chistie** [comment]
I was able to get the errors after changing the types in the files:

.tsx file:
20error1

.stories.tsx file:
20error2

**Comment by Mihan786Chistie** [comment]
@lucasbordeau Any updates?

**Comment by FelixMalfait** [comment]
Thanks!
I was about to merge this but I realize this might build tests/stories into the prod app (?).
One solution would be to do environment-specific tsconfig.json (possible to avoid copy/paste through inheritance)
Let's wait for Lucas to decide when he comes back next week

**Comment by lucasbordeau** [comment]
@Mihan786Chistie Félix is right, let's create a new tsconfig.dev.json where we'll put the config that includes everything.

We should then use the variable `isBuildCommand` in vite config to determine which tsconfig we should use.

**Comment by lucasbordeau** [comment]
@Mihan786Chistie I've made the required changes, there's only twenty-ui stories that are still left unchecked by twenty-front dev command, I'll fix it and it'll be done.

**Comment by Mihan786Chistie** [comment]
@lucasbordeau Thanks for the help. Will look forward for other issues

**Review by lucasbordeau — APPROVED** [review]

**Comment by lucasbordeau** [comment]
@Mihan786Chistie I'm assigning you on this issue : https://github.com/twentyhq/twenty/issues/4871

**Pull request merged**